### PR TITLE
SettingsList.py clean-up & hell mode preset update

### PR DESCRIPTION
**Pull request by ETR-BTF**
A revival of #1310

* Reorders the settings in SettingsList.py to make the Spoiler Log more readable for Support and users.
** The order follows the order in which settings appear in the GUI for about 95%. The exception is that certain major settings are bumped to the top of the settings list, such as ALR, rainbow bridge, gbk shuffle, triforce hunt.
* Fixes a typo in the Presets tooltip.
* Modifies the Adult/Child Link Models tooltips.
* Enables Shuffle Frog Song Rupees in the Hell Mode preset.
* Edit: Reorders the Glitchless Tricks.
* Edit: MQ tags are now also added to the tricks relevant for MQ, as the search feature doesn't find them all properly.

~~This PR has not (yet) touched the order of Glitchless Tricks. If that's something people would like to see then I'm willing to expand this PR to include that, but I think we should discuss a desired order first.~~

Edit: Unit tests passed and I was able to generate a seed + log on Hell Mode.

### CODE REVIEW

File: SettingsList.py [hunk 1]
```diff
@@ -1671,8 +1671,11 @@ def __init__(self, name, gui_text, min, max, default, step=1,
                     '''},
 }
 
+# Below is the list of possible settings.
+# They are mostly isted in the order in which they appear in the GUI
```
Comment (Zannick) [inline SettingsList.py]:
```suggestion
# They are mostly listed in the order in which they appear in the GUI
```

File: SettingsList.py [hunk 2]
```diff
@@ -2507,782 +2630,902 @@ def __init__(self, name, gui_text, min, max, default, step=1,
...
-        ''',
-        shared         = True,
-    ),
-    Checkbutton(
-        name           = 'fast_chests',
-        gui_text       = 'Fast Chest Cutscenes',
-        gui_tooltip    = '''\
-            All chest animations are fast. If disabled,
-            the animation time is slow for major items.
+            Try different combination out, such as:
```
Comment (Zannick) [inline SettingsList.py]:
```suggestion
            Try different combinations out, such as:
```

**Comment by ETR-BTF** [comment]
Converting to draft because I do plan to include the tricks into this, and we're also still discussing what the worst settings for Hell Mode would be in Discord. I might finish these things by the end of the day

**Comment by ETR-BTF** [comment]
It genned a Hell Mode seed and passed the unit tests, and a quick count suggests I didn't accidentally delete a trick lol so that's good.

**Comment by r0bd0g** [comment]
When a trick applies to both variants it seems really inconsistent about whether the MQ tag is given.

**Comment by ETR-BTF** [comment]
> When a trick applies to both variants it seems really inconsistent about whether the MQ tag is given.

That's because I only searched for "MQ", but some tricks spell out "Master Quest" explicitly. I should have gotten them all now, but please let me know if I missed any.

**Comment by r0bd0g** [comment]
Technically, logic_fewer_tunic_requirements applies to both, but that one probably doesn't need it called out.

logic_dc_jump
logic_child_deadhand
logic_forest_vines

Except forest_vines, which was added together with MQ (as part of 'various advanced tricks'), all of those predate MQ. I would say child_deadhand though probably still doesn't need that it applies to both called out in the tooltip.

You added whitespace line 1228.

Torn on whether logic_dc_hammer_floor should have it. (logic_shadow_bongo currently doesn't, but that one doesn't have a special case where it can be relevant in some settings only in MQ the way the DC boss room does.)

**Comment by ETR-BTF** [comment]
I think it makes sense to add the tag for the DC hammer trick but not for Bongo or Deadhand. I've also added it to the FTR trick and added some more information to the tooltip just for clarification. I don't think it's obvious that the trick applies to the same areas for the MQ versions.

**Review by cjohnson57 — APPROVED** [review]

**Pull request merged**